content/docs/user-guide/data-pipelines/defining-pipelines.md

```diff
@@ -0,0 +1,253 @@
+# Defining Pipelines
+
+Pipelines represent data workflows that you want to **reproduce** reliably -- so
+the results are consistent. The regular pipelining process involves:
+
+- Obtain and `dvc add` the project's initial data requirements (see [Data
+  Management]). This <abbr>caches</abbr> the data and generates `.dvc` files.
+
+- Define the pipeline [stages](#stages) in `dvc.yaml` files (more details
+  later). Example structure:
+
+  ```yaml
+  stages:
+    prepare: ... # stage 1 definition
+    train: ... # stage 2 definition
+    evaluate: ... # stage 3 definition
+  ```
+
+- Capture other useful metadata such as runtime
+  [parameters](#parameter-dependencies), performance [metrics], and [plots] to
+  visualize. DVC supports multiple file formats for these.
+
+<admon type="info">
+
+We call this file-based definition _codification_ (YAML format in our case). It
+has the added benefit of allowing you to develop pipelines on standard Git
+workflows ([GitOps]).
+
+[gitops]: /doc/use-cases/versioning-data-and-model-files
+
+</admon>
+
+Stages usually take some data and run some code, producing an output (e.g. an ML
+model). The pipeline is formed by making them interdependent, meaning that the
+output of a stage becomes the input of another, and so on. Technically, this is
+called a _dependency graph_ (DAG).
+
+[data management]: /doc/start/data-management
+[metrics]: /doc/command-reference/metrics
+[plots]: /doc/user-guide/visualizing-plots
+
+## Directed Acyclic Graph (DAG)
```

Comment:
can be an expandable section?

Reply:
Not sure. It's a pretty key concept like you mentioned some time ago and it's only one paragraph of theory, the other text in this section includes relevant guidance.

Reply:
yes, but it contains very technical information to my mind

Reply:
wikipedia style :)

Reply:
OK well no strong opinion, since all the links from other pages direct to that section will expand it anyway. Changed and moved the other details to the Stages section ().

2 minor questions though:
- Should it stay an H2 so it's visible in the right ToC sidebar?
- The internal link (from within this same page) to that section does not expand it, is that fine?

Reply:
c9e627222